### PR TITLE
Meta og:url must be a full url

### LINKED ISSUE #179: og:url in boilerplate is incorrect
As per discusstion in https://github.com/MoOx/statinamic/issues/104#issuecomment-184322087

`og:url` must be a absolute path.

This is doable by joining `__url` with `pkg.homepage` using join-uri utility. But you marked it as private API. Any thought ?

### PR DESCRIPTION
**Pull request by thangngoc89**
Close #179

### CODE REVIEW

File: src/builder/webpack/config.common.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ const wrap = JSON.stringify
 
 export default (config) => {
   const { webpackConfig } = config
+  const userUrl = pkg.homepage.replace(/\/?$/, "/") // ensure trailing slash
```
Comment (MoOx) [inline src/builder/webpack/config.common.js]: you should use config.baseUrl object.

File: boilerplate/web_modules/layouts/Page/index.js [hunk 2]
```diff
@@ -25,11 +25,14 @@ class Page extends Component {
     )
 
     const metaTitle = head.metaTitle ? head.metaTitle : head.title
+    // __url has surrounding slashes so we need to remove trailing slash
+    // PHENOMIC_USER_URL
+    const fullUrl = process.env.PHENOMIC_USER_URL.replace(/\/?$/, "") + __url
```
Comment (MoOx) [inline boilerplate/web_modules/layouts/Page/index.js]: This is not pretty, we need something prettier :)

Comment (thangngoc89) [inline boilerplate/web_modules/layouts/Page/index.js]: join-uri utility ?

**Comment by thangngoc89** [comment]
PR updated. I exposed joinUri in `src/index.js`. This is a better way, we don't need to change anything internally. If we want to make an API public, we only need to do the same.

File: boilerplate/web_modules/layouts/Page/index.js [hunk 3]
```diff
@@ -25,11 +26,12 @@ class Page extends Component {
     )
 
     const metaTitle = head.metaTitle ? head.metaTitle : head.title
+    const fullUrl = joinUri(process.env.PHENOMIC_USER_URL, __url)
```
Comment (MoOx) [inline boilerplate/web_modules/layouts/Page/index.js]: Can you avoid the const and pu the join directly the right place?

File: src/index.js [hunk 4]
```diff
@@ -1,3 +1 @@
-throw new Error(
-  "Please use submodules directly (e.g.: 'phenomic/lib/client')"
-)
+export joinUri from "./_utils/join-uri"
```
Comment (MoOx) [inline src/index.js]: good idea.

Comment (MoOx) [inline src/index.js]: with webpack 2, we might exposes everything in "phenomic".

**Comment by thangngoc89** [comment]
PR updated

File: src/builder/webpack/config.common.js [hunk 5]
```diff
@@ -21,7 +22,7 @@ export default (config) => {
         ),
 
         PHENOMIC_USER_PATHNAME: wrap(process.env.PHENOMIC_USER_PATHNAME),
-
+        PHENOMIC_USER_URL: wrap(url.format(config.baseUrl)),
```
Comment (MoOx) [inline src/builder/webpack/config.common.js]: can you add it into the doc as well? I documented other values https://github.com/MoOx/phenomic/blob/4dc38cffb8cc37cd1747297106907e2c2ce759e3/docs/content/docs/usage/scripting.md#constants

Comment (thangngoc89) [inline src/builder/webpack/config.common.js]: Added PHENOMIC_USER_URL to docs

**Comment by MoOx** [comment]
👍 
I won't bother you to document joinUrl but we might do this someday. I opened #433 for this.

**Comment by MoOx** [comment]
We should also add the new const in the CHANGELOG next to the others.

**Comment by thangngoc89** [comment]
I updated CHANGELOG

**Pull request merged**